Update protocol buffer dependencies

sergefdrv opened the pull request:
<!-- NOTE: Please check the contribution guideline before submitting -->
 
<!-- describe the purpose of the pull request, as well as its benefits
     and possible concerns related to the proposed changes -->
This pull request fully updates the code to use Go protocol buffers APIv2. This will help to regenerate Protobuf code when introducing new message types.

<!-- and reference any issues resolved by the pull request, e.g. #123 -->

Commit:
Update Go module dependencies

Update all dependencies with:

    go get -u ./...
    go mod tidy

Signed-off-by: Sergey Fedorov <sergey.fedorov@neclab.eu>

Commit:
Use Go protocol buffers APIv2

At this point completely stop using the old Go protocol buffers APIv1
provided by github.com/golang/protobuf package.

Signed-off-by: Sergey Fedorov <sergey.fedorov@neclab.eu>

Commit:
Adjust Protobuf package names

Make the package names more specific.

Signed-off-by: Sergey Fedorov <sergey.fedorov@neclab.eu>

sergefdrv commented:
CI with Go 1.11 has failed because some updated dependency requires at least Go 1.12. Maybe it's time to drop support for Go 1.11 since [current stable versions](https://golang.org/dl/) of Go are 1.15 and 1.16.

Commit:
Update supported versions of Go

Replace support for Go 1.11 with 1.13 and add 1.16. The current stable
versions are 1.15 and 1.16, whereas Ubuntu 20.04 LTS provides Go 1.13.

Signed-off-by: Sergey Fedorov <sergey.fedorov@neclab.eu>

ynamiki commented on messages/protobuf/pb/messages.proto:
```diff
@@ -18,7 +18,7 @@
 syntax = "proto3";
 
 // Package pb defines MinBFT protocol messages in Protobuf format.
```
This comment also should be updated: "Package *pb* defines ..."

sergefdrv replied:
I guess we'd better omit the package name from the comment since the generated Go code still declares the package name as `pb`.

sergefdrv commented:
Summary of [changes](https://github.com/hyperledger-labs/minbft/compare/3c36174f9bd0339457199e46501228c9c7634c48..d7ec186aa6572fdb23c767c66dd7c689c3ae9341):
 - Added support for Go v1.13
 - Fixed comment in `messages.proto`

sergefdrv commented:
Summary of [changes](https://github.com/hyperledger-labs/minbft/compare/d7ec186aa6572fdb23c767c66dd7c689c3ae9341..c251ef1fd5f0f9fef1863dbf09f94d6fc897c7a2): adjusted commit message about supported Go versions.

sergefdrv commented:
Summary of [changes](https://github.com/hyperledger-labs/minbft/compare/c251ef1fd5f0f9fef1863dbf09f94d6fc897c7a2..d703b24df6a55b72b3037fa0ee8540daadd341e2): regenerated Protobuf code.

ynamiki reviewed (approved):
Thanks!
